### PR TITLE
feat(translator): Implement Retry in BTP

### LINKED ISSUE #1907: Support retries
*Description*:

Add retry settings to be able to retry making requests to a backend

Relates to
* https://github.com/kubernetes-sigs/gateway-api/issues/1731
* https://github.com/envoyproxy/gateway/issues/1821#issuecomment-1729935509
* https://github.com/envoyproxy/gateway/pull/1899#issuecomment-1725910504

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

### PR DESCRIPTION
**Pull request by guydc**
**What this PR does / why we need it**:
Implements retry API in BTP: #2657

**Which issue(s) this PR fixes**:
Fixes #1907

### CODE REVIEW

File: internal/xds/translator/testdata/in/xds-ir/retry.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  port: 10080
+  hostnames:
+  - "*"
+  path:
+    mergeSlashes: true
+    escapedSlashesAction: UnescapeAndRedirect
+  routes:
+  - name: "first-route"
+    hostname: "*"
+    retry:
```
Comment (arkodg) [inline internal/xds/translator/testdata/in/xds-ir/retry.yaml:13]: can we add another case with a empty `retry: {}` to its easier to cross check defaults in the future, thanks

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by shawnh2 — APPROVED** [review]
lgtm, thanks

**Pull request merged**